### PR TITLE
Bug 2033167: oc extract: Create target directory if not exist

### PR DESCRIPTION
**Pull request by ardaguclu**
Currently, `oc extract` command returns error, if target directory is not existed. This PR creates all nested directories prior to extract data.

This will easen usage of `oc extract` command.

### CODE REVIEW

File: pkg/cli/extract/extract.go [hunk 1]
```diff
@@ -105,19 +105,24 @@ func (o *ExtractOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args [
 		return err
 	}
 
+	if o.TargetDirectory != "-" {
```
Comment (ingvagabund) [inline pkg/cli/extract/extract.go:108]: I'd prefer to move creation of the target directory as close to the line of code where it is needed. E.g. https://github.com/openshift/oc/blob/master/pkg/cli/extract/extract.go#L167

Comment (ingvagabund) [reply pkg/cli/extract/extract.go:108]: When creating the target directory right before it is needed one might get "permission denied". Knowing this in advance would help to save the entire extraction step on the other hand. Which makes the current change more preferable on the other hand.

**Comment by ardaguclu** [comment]
e2e-agnostic-cmd failure is real
/hold

**Review by soltysh — CHANGES_REQUESTED** [review]

**Comment by ardaguclu** [comment]
None of the failures is related to this patch;

/retest-required
/hold cancel

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**